src/rpc/connectivity/cluster.cc

```diff
@@ -380,17 +383,18 @@ void connectivity_cluster_t::run_t::connect_to_peer(
     rate_control->unlock(1);
 }
 
-void connectivity_cluster_t::run_t::join_blocking(
+join_result_t connectivity_cluster_t::run_t::join_blocking(
         const peer_address_t peer,
         boost::optional<peer_id_t> expected_id,
+        boost::optional<server_id_t> expected_server_id,
         const int join_delay_secs,
         auto_drainer_t::lock_t drainer_lock) THROWS_NOTHING {
     drainer_lock.assert_is_holding(&parent->current_run->drainer);
     parent->assert_thread();
     {
         mutex_assertion_t::acq_t acq(&attempt_table_mutex);
         if (attempt_table.find(peer) != attempt_table.end()) {
-            return;
+            return join_result_t::PERMANENT_ERROR;
```

Comment:
@danielmewes I think if an attempt is already occurring then maybe this is a `TEMPORARY_ERROR` ?